src/ClientResolver.php

```diff
@@ -917,17 +926,41 @@ public static function _apply_http_handler($value, array &$args, HandlerList $li
         );
     }
 
+    public static function _apply_app_id($inputAppId, array &$args)
+    {
+        // AppId should not be longer than 50 chars
+        static $MAX_APP_ID_LENGTH = 50;
+        if (strlen($inputAppId) > $MAX_APP_ID_LENGTH) {
+            trigger_error("The provided or configured value for AppId, an user agent parameter, exceeds the maximum length of $MAX_APP_ID_LENGTH characters.", E_USER_WARNING);
+        }
+
+        $args['app_id'] = $inputAppId;
+    }
+
+    public static function _default_app_id(array $args)
+    {
+        return ConfigurationResolver::resolve(
+            'sdk_ua_app_id',
+            '',
```

Comment:
Not familiar with the spec— the default is an empty string?

Reply:
There is not a default value for app_id. If empty is returned here then, that means customer did not provided or configured an app_id value, and in that case this parameter is just ignored and not appended to the user agent header.